src/Engine.sol

```diff
@@ -63,50 +73,78 @@ contract Engine is
     ISpotMarketProxy internal immutable SPOT_MARKET_PROXY;
 
-    /// @notice Synthetix v3 sUSD contract
+    /// @notice Synthetix v3 $sUSD contract
     IERC20 internal immutable SUSD;
 
     /*//////////////////////////////////////////////////////////////
                                  STATE
     //////////////////////////////////////////////////////////////*/
 
-    /// @notice bit mapping that stores whether a conditional order nonce has been executed
-    /// @dev nonce is specific to the account id associated with the conditional order
+    /// @notice bit mapping that stores whether a conditional
+    /// order nonce has been executed
+    /// @dev nonce is specific to the account id associated
+    /// with the conditional order
     mapping(uint128 accountId => mapping(uint256 index => uint256 bitmap))
         public nonceBitmap;
 
-    /// @notice mapping of account id to sUSD balance (i.e. credit available to pay for fee(s))
-    /// @dev sUSD can be credited to the Engine to pay for fee(s)
+    /// @notice mapping of account id to $sUSD balance
+    /// (i.e. credit available to pay for fee(s))
+    /// @dev $sUSD can be credited to the Engine to pay for fee(s)
     mapping(uint128 accountId => uint256) public credit;
 
     /*//////////////////////////////////////////////////////////////
                               CONSTRUCTOR
     //////////////////////////////////////////////////////////////*/
 
     /// @notice Constructs the Engine contract
-    /// @param _perpsMarketProxy Synthetix v3 perps market proxy contract
-    /// @param _spotMarketProxy Synthetix v3 spot market proxy contract
-    /// @param _sUSDProxy Synthetix v3 sUSD contract
-    /// @param _pDAO Kwenta owned/operated multisig address that can authorize upgrades
+    /// @dev Zap constructor will revert if any of the
+    /// addresses are zero
+    /// @param _perpsMarketProxy Synthetix v3 perps
+    /// market proxy contract
+    /// @param _spotMarketProxy Synthetix v3 spot
+    /// market proxy contract
+    /// @param _sUSDProxy Synthetix v3 $sUSD contract
+    /// @param _pDAO Kwenta owned/operated multisig address
+    /// that can authorize upgrades
+    /// @param _usdc $USDC token contract address
+    /// @param _sUSDCId Synthetix v3 Spot Market ID for $sUSDC
     /// @custom:oz-upgrades-unsafe-allow constructor
     constructor(
         address _perpsMarketProxy,
         address _spotMarketProxy,
         address _sUSDProxy,
-        address _pDAO
-    ) {
+        address _pDAO,
+        address _usdc,
+        uint128 _sUSDCId
+    ) Zap(_usdc, _sUSDProxy, _spotMarketProxy, _sUSDCId) {
         if (_perpsMarketProxy == address(0)) revert ZeroAddress();
-        if (_spotMarketProxy == address(0)) revert ZeroAddress();
-        if (_sUSDProxy == address(0)) revert ZeroAddress();
 
         PERPS_MARKET_PROXY = IPerpsMarketProxy(_perpsMarketProxy);
         SPOT_MARKET_PROXY = ISpotMarketProxy(_spotMarketProxy);
```

Comment:
**Optimization:** `SPOT_MARKET_PROXY` is stored also in `Zap` under `_SPOT_MARKET_PROXY`, so deployed bytecode could be reduced by only using the `Zap` version.

Reply:
fixed